src/Host-SDK/Temporal.Worker.Workflows/WorkflowsApi.cs

```diff
@@ -0,0 +1,309 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Temporal.Common.WorkflowConfiguration;
+using Temporal.Common.DataModel;
+using Temporal.Serialization;
+
+namespace Temporal.Worker.Workflows
+{
+    public class WorkflowsApi
+    {
+    }
+
+    // ----------- -----------
+
+    public class WorkflowContext
+    {        
+        public IWorkflowExecutionConfiguration WorkflowExecutionConfig { get; }
+        public IWorkflowImplementationConfiguration WorkflowImplementationConfig { get; }
+        public IOrchestrationService Orchestrator { get; }
+        public WorkflowRunContext CurrentRun { get; }
+        public WorkflowPreviousRunContext  LastRun { get; }
+        public IDeterministicApi DeterministicApi { get; set; }
+
+        /// <summary>Get the serializer for the specified payload.
+        /// If metadata specifies an available serializer - get that one;
+        /// If metadata specifies an unavailable serializer - throw;
+        /// If metadata specified nothing - get the default form the config.</summary>        
+        public IPayloadSerializer GetSerializer(PayloadsCollection payloads) { return null; }
```

Comment:
Why invent a new name instead of using the `DataConverter` interface we have in the other SDKs?

Reply:
Also this isn't exposed in the other SDKs, why is it required here?

Reply:
> Why invent a new name instead of using the DataConverter interface we have in the other SDKs?
 Serializer is a common .NET term. However, I do not mind either way, so happy to rename it.

Reply:
> Also this isn't exposed in the other SDKs, why is it required here?
It can be set via config. 
I'll add a clarifying example.

Reply:
> Serializer is a common .NET term. However, I do not mind either way, so happy to rename it.

I think we should. Serializer is also the common Java (and Rust) term, and Go uses Marshaler, but "DataConverter" is still chosen. I've found when it comes to SDK parity and parlance, the more similarities without overly breaking per-lang norms, the easier it is to talk about capabilities in a general way.